core/arch/arm/kernel/thread.c

```diff
@@ -152,7 +152,15 @@ thread_pm_handler_t thread_system_reset_handler_ptr;
 static vaddr_t thread_user_kcode_va;
 long thread_user_kcode_offset;
 static size_t thread_user_kcode_size;
-#endif /*CFG_CORE_UNMAP_CORE_AT_EL0*/
+#endif
+
+#if defined(CFG_CORE_UNMAP_CORE_AT_EL0) && \
+    defined(CFG_CORE_WORKAROUND_SPECTRE_BP_SEC) && defined(ARM64)
+long thread_user_kdata_sp_offset;
+static uint8_t thread_user_kdata_page[
+	ROUNDUP(sizeof(thread_core_local), SMALL_PAGE_SIZE)]
+	__aligned(SMALL_PAGE_SIZE) __section(".nozi.kdata_page");
```

Comment:
All active threads uses the same `thread_user_kdata_page`?

Reply:
It's actually the cores that uses the same `thread_user_kdata_page`.